_posts/2020-10-05-Evidence_locker_overview.md

```diff
@@ -0,0 +1,98 @@
+---
+layout: post
+category: Architecture
+title: Evidence locker overview
+---
+
+At the core of the Auditree framework is a [Git][] repository known as the
+evidence locker.  Each Auditree execution environment requires a dedicated
+evidence locker in order to execute properly.  Serving as the Auditree
+data store, the evidence locker houses evidence gathered by fetchers,
+evidence generated by checks and/or evidence planted by the [Auditree Plant][]
+tool.
+
+## Local vs. remote
+
+The reasoning behind our choice of [Git][] as the Auditree data store is
+outlined nicely in our [previous post][] but in a nutshell, a local evidence
+locker works hand-in-hand with a remote evidence locker hosted by any Git
+repository hosting service (GitHub, GitLab, ...).  Although it is possible to
+execute Auditree with only a local evidence locker, a typical implementation of
+an Auditree execution environment will pull down a local clone of a remote
+evidence locker, perform Auditree operations, store the results of those
+operations as evidence in the clone (local locker) and once complete, push
+changes back to the remote evidence locker.
+
+## Organization
+
+An evidence locker is organized into (up to) four top level folders:
+
+- **raw**: All evidence retrieved by fetchers is stored here.
+- **reports**: All report evidence generated by checks is stored here.
+- **external**: All evidence gathered or generated outside of the Auditree
+framework and planted by [Auditree Plant][] is stored here.
+- **notifications**: All check notifications sent to the locker are stored
+here.
+
+## OK, but what makes it a locker?
+
+...in a word, **metadata**.
+
+As noted earlier, all files are organized according to their file type.  But
+the Auditree framework still needs a way to treat these files as evidence and
+that's where the metadata comes in.  The `raw`, `reports` and `external`
+evidence top level folders are purposefully shallow.  Each folder can be
+organized into one or more categories of evidence sub-folders.  Each evidence
+category sub-folder contains an `index.json` file that stores all of the
+metadata for evidences within that category.
+
+Evidence metadata includes:
+
+- Basic evidence metadata
+   - An evidence last update date/time
+   - An evidence time to live duration
+   - An evidence description
+- Partitioned (raw) evidence metadata
+   - Basic evidence metadata
+   - Fields used to partition the evidence
+   - A mapping of evidence partition to evidence file
+- Report evidence metadata
+   - Basic evidence metadata
+   - Checks used to generate the report
+   - A list of evidence used by the checks to produce the report
+- External evidence metadata
+   - Basic evidence metadata
+   - Planter email
+- Pruned evidence metadata
+   - Basic/Partitioned/Report/External (whichever applies)
+   - Pruner email
+   - Evidence tombstone including an end of life date/time, last update
+   date/time, and a reason for pruning.
+
```

Comment:
Worth mentioning that we're duplicating some of the git file metadata here, and the reason why (if the file hasn't changed the git metadata won't update but index.json will).

Reply:
Done in https://github.com/auditree/auditree.github.io/pull/6/commits/406f0ed0a3b7b1b75e25c945c7f52be953d95e0f